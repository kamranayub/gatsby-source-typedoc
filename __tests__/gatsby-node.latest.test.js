const { helpers, cleanNodeForSnapshot } = require("./__fixtures__/test-utils");
const { sourceNodes } = require("../gatsby-node");

describe("gatsby-node: sourceNodes", () => {
  describe("typedoc: latest", () => {
    it("should generate project", async () => {
      let typedocNode;

      helpers.actions.createNode.mockImplementation((node) => {
        typedocNode = node;
      });

      await sourceNodes(helpers, {
        src: [require.resolve("./__fixtures__/simple/index.ts")],
        typedoc: {
          logLevel: "Verbose",
          disableSources: true,
          tsconfig: require.resolve("./__fixtures__/simple/tsconfig.json"),
        },
      });

      expect(typedocNode).toBeDefined();
      cleanNodeForSnapshot(typedocNode);
      expect(typedocNode).toMatchSnapshot();
    });
  });
});
