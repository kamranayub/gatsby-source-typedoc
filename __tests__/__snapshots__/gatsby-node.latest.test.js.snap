// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-node: sourceNodes typedoc: latest should generate project 1`] = `
{
  "id": "typedoc-default",
  "internal": {
    "content": "{"id":0,"name":"default","kind":1,"kindString":"Project","flags":{},"children":[{"id":3,"name":"TestClass1","kind":128,"kindString":"Class","flags":{},"children":[{"id":4,"name":"constructor","kind":512,"kindString":"Constructor","flags":{},"signatures":[{"id":5,"name":"new TestClass1","kind":16384,"kindString":"Constructor signature","flags":{},"type":{"type":"reference","id":3,"name":"TestClass1"}}]},{"id":6,"name":"propPrivate","kind":1024,"kindString":"Property","flags":{"isPrivate":true},"type":{"type":"intrinsic","name":"string"}},{"id":7,"name":"propPublic","kind":1024,"kindString":"Property","flags":{"isPublic":true},"type":{"type":"intrinsic","name":"string"}},{"id":14,"name":"getterPublic","kind":262144,"kindString":"Accessor","flags":{},"getSignature":{"id":15,"name":"getterPublic","kind":524288,"kindString":"Get signature","flags":{},"type":{"type":"intrinsic","name":"number"}}},{"id":16,"name":"setterPublic","kind":262144,"kindString":"Accessor","flags":{},"setSignature":{"id":17,"name":"setterPublic","kind":1048576,"kindString":"Set signature","flags":{},"parameters":[{"id":18,"name":"value","kind":32768,"kindString":"Parameter","flags":{},"type":{"type":"intrinsic","name":"string"}}],"type":{"type":"intrinsic","name":"void"}}},{"id":10,"name":"getPrivate","kind":2048,"kindString":"Method","flags":{"isPrivate":true},"signatures":[{"id":11,"name":"getPrivate","kind":4096,"kindString":"Call signature","flags":{},"type":{"type":"intrinsic","name":"boolean"}}]},{"id":12,"name":"getProtected","kind":2048,"kindString":"Method","flags":{"isProtected":true},"signatures":[{"id":13,"name":"getProtected","kind":4096,"kindString":"Call signature","flags":{},"type":{"type":"intrinsic","name":"boolean"}}]},{"id":8,"name":"getPublic","kind":2048,"kindString":"Method","flags":{"isPublic":true},"signatures":[{"id":9,"name":"getPublic","kind":4096,"kindString":"Call signature","flags":{},"type":{"type":"intrinsic","name":"boolean"}}]}],"groups":[{"title":"Constructors","children":[4]},{"title":"Properties","children":[6,7]},{"title":"Accessors","children":[14,16]},{"title":"Methods","children":[10,12,8]}]},{"id":1,"name":"exportedFn","kind":64,"kindString":"Function","flags":{},"signatures":[{"id":2,"name":"exportedFn","kind":4096,"kindString":"Call signature","flags":{},"type":{"type":"intrinsic","name":"number"}}]},{"id":19,"name":"util","kind":64,"kindString":"Function","flags":{},"signatures":[{"id":20,"name":"util","kind":4096,"kindString":"Call signature","flags":{},"type":{"type":"intrinsic","name":"string"}}]}],"groups":[{"title":"Classes","children":[3]},{"title":"Functions","children":[1,19]}]}",
    "contentDigest": "testdigest",
    "type": "Typedoc",
  },
  "source": {
    "categories": undefined,
    "children": [
      {
        "categories": undefined,
        "children": [
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {},
            "getSignature": undefined,
            "groups": undefined,
            "id": 4,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 512,
            "kindString": "Constructor",
            "name": "constructor",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": [
              {
                "comment": undefined,
                "flags": {},
                "id": 5,
                "implementationOf": undefined,
                "inheritedFrom": undefined,
                "kind": 16384,
                "kindString": "Constructor signature",
                "name": "new TestClass1",
                "originalName": undefined,
                "overwrites": undefined,
                "parameters": undefined,
                "type": {
                  "externalUrl": undefined,
                  "id": 3,
                  "name": "TestClass1",
                  "type": "reference",
                  "typeArguments": undefined,
                },
                "typeParameter": undefined,
              },
            ],
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {
              "isPrivate": true,
            },
            "getSignature": undefined,
            "groups": undefined,
            "id": 6,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 1024,
            "kindString": "Property",
            "name": "propPrivate",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": undefined,
            "sources": undefined,
            "type": {
              "name": "string",
              "type": "intrinsic",
            },
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {
              "isPublic": true,
            },
            "getSignature": undefined,
            "groups": undefined,
            "id": 7,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 1024,
            "kindString": "Property",
            "name": "propPublic",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": undefined,
            "sources": undefined,
            "type": {
              "name": "string",
              "type": "intrinsic",
            },
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {},
            "getSignature": {
              "comment": undefined,
              "flags": {},
              "id": 15,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 524288,
              "kindString": "Get signature",
              "name": "getterPublic",
              "originalName": undefined,
              "overwrites": undefined,
              "parameters": undefined,
              "type": {
                "name": "number",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
            "groups": undefined,
            "id": 14,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 262144,
            "kindString": "Accessor",
            "name": "getterPublic",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": undefined,
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {},
            "getSignature": undefined,
            "groups": undefined,
            "id": 16,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 262144,
            "kindString": "Accessor",
            "name": "setterPublic",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": {
              "comment": undefined,
              "flags": {},
              "id": 17,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 1048576,
              "kindString": "Set signature",
              "name": "setterPublic",
              "originalName": undefined,
              "overwrites": undefined,
              "parameters": [
                {
                  "comment": undefined,
                  "defaultValue": undefined,
                  "flags": {},
                  "id": 18,
                  "kind": 32768,
                  "kindString": "Parameter",
                  "name": "value",
                  "originalName": undefined,
                  "type": {
                    "name": "string",
                    "type": "intrinsic",
                  },
                },
              ],
              "type": {
                "name": "void",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
            "signatures": undefined,
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {
              "isPrivate": true,
            },
            "getSignature": undefined,
            "groups": undefined,
            "id": 10,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 2048,
            "kindString": "Method",
            "name": "getPrivate",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": [
              {
                "comment": undefined,
                "flags": {},
                "id": 11,
                "implementationOf": undefined,
                "inheritedFrom": undefined,
                "kind": 4096,
                "kindString": "Call signature",
                "name": "getPrivate",
                "originalName": undefined,
                "overwrites": undefined,
                "parameters": undefined,
                "type": {
                  "name": "boolean",
                  "type": "intrinsic",
                },
                "typeParameter": undefined,
              },
            ],
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {
              "isProtected": true,
            },
            "getSignature": undefined,
            "groups": undefined,
            "id": 12,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 2048,
            "kindString": "Method",
            "name": "getProtected",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": [
              {
                "comment": undefined,
                "flags": {},
                "id": 13,
                "implementationOf": undefined,
                "inheritedFrom": undefined,
                "kind": 4096,
                "kindString": "Call signature",
                "name": "getProtected",
                "originalName": undefined,
                "overwrites": undefined,
                "parameters": undefined,
                "type": {
                  "name": "boolean",
                  "type": "intrinsic",
                },
                "typeParameter": undefined,
              },
            ],
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
          {
            "categories": undefined,
            "children": undefined,
            "comment": undefined,
            "defaultValue": undefined,
            "extendedBy": undefined,
            "extendedTypes": undefined,
            "flags": {
              "isPublic": true,
            },
            "getSignature": undefined,
            "groups": undefined,
            "id": 8,
            "implementationOf": undefined,
            "implementedBy": undefined,
            "implementedTypes": undefined,
            "indexSignature": undefined,
            "inheritedFrom": undefined,
            "kind": 2048,
            "kindString": "Method",
            "name": "getPublic",
            "originalName": undefined,
            "overwrites": undefined,
            "setSignature": undefined,
            "signatures": [
              {
                "comment": undefined,
                "flags": {},
                "id": 9,
                "implementationOf": undefined,
                "inheritedFrom": undefined,
                "kind": 4096,
                "kindString": "Call signature",
                "name": "getPublic",
                "originalName": undefined,
                "overwrites": undefined,
                "parameters": undefined,
                "type": {
                  "name": "boolean",
                  "type": "intrinsic",
                },
                "typeParameter": undefined,
              },
            ],
            "sources": undefined,
            "type": undefined,
            "typeParameters": undefined,
          },
        ],
        "comment": undefined,
        "defaultValue": undefined,
        "extendedBy": undefined,
        "extendedTypes": undefined,
        "flags": {},
        "getSignature": undefined,
        "groups": [
          {
            "categories": undefined,
            "children": [
              4,
            ],
            "title": "Constructors",
          },
          {
            "categories": undefined,
            "children": [
              6,
              7,
            ],
            "title": "Properties",
          },
          {
            "categories": undefined,
            "children": [
              14,
              16,
            ],
            "title": "Accessors",
          },
          {
            "categories": undefined,
            "children": [
              10,
              12,
              8,
            ],
            "title": "Methods",
          },
        ],
        "id": 3,
        "implementationOf": undefined,
        "implementedBy": undefined,
        "implementedTypes": undefined,
        "indexSignature": undefined,
        "inheritedFrom": undefined,
        "kind": 128,
        "kindString": "Class",
        "name": "TestClass1",
        "originalName": undefined,
        "overwrites": undefined,
        "setSignature": undefined,
        "signatures": undefined,
        "sources": undefined,
        "type": undefined,
        "typeParameters": undefined,
      },
      {
        "categories": undefined,
        "children": undefined,
        "comment": undefined,
        "defaultValue": undefined,
        "extendedBy": undefined,
        "extendedTypes": undefined,
        "flags": {},
        "getSignature": undefined,
        "groups": undefined,
        "id": 1,
        "implementationOf": undefined,
        "implementedBy": undefined,
        "implementedTypes": undefined,
        "indexSignature": undefined,
        "inheritedFrom": undefined,
        "kind": 64,
        "kindString": "Function",
        "name": "exportedFn",
        "originalName": undefined,
        "overwrites": undefined,
        "setSignature": undefined,
        "signatures": [
          {
            "comment": undefined,
            "flags": {},
            "id": 2,
            "implementationOf": undefined,
            "inheritedFrom": undefined,
            "kind": 4096,
            "kindString": "Call signature",
            "name": "exportedFn",
            "originalName": undefined,
            "overwrites": undefined,
            "parameters": undefined,
            "type": {
              "name": "number",
              "type": "intrinsic",
            },
            "typeParameter": undefined,
          },
        ],
        "sources": undefined,
        "type": undefined,
        "typeParameters": undefined,
      },
      {
        "categories": undefined,
        "children": undefined,
        "comment": undefined,
        "defaultValue": undefined,
        "extendedBy": undefined,
        "extendedTypes": undefined,
        "flags": {},
        "getSignature": undefined,
        "groups": undefined,
        "id": 19,
        "implementationOf": undefined,
        "implementedBy": undefined,
        "implementedTypes": undefined,
        "indexSignature": undefined,
        "inheritedFrom": undefined,
        "kind": 64,
        "kindString": "Function",
        "name": "util",
        "originalName": undefined,
        "overwrites": undefined,
        "setSignature": undefined,
        "signatures": [
          {
            "comment": undefined,
            "flags": {},
            "id": 20,
            "implementationOf": undefined,
            "inheritedFrom": undefined,
            "kind": 4096,
            "kindString": "Call signature",
            "name": "util",
            "originalName": undefined,
            "overwrites": undefined,
            "parameters": undefined,
            "type": {
              "name": "string",
              "type": "intrinsic",
            },
            "typeParameter": undefined,
          },
        ],
        "sources": undefined,
        "type": undefined,
        "typeParameters": undefined,
      },
    ],
    "comment": undefined,
    "flags": {},
    "groups": [
      {
        "categories": undefined,
        "children": [
          3,
        ],
        "title": "Classes",
      },
      {
        "categories": undefined,
        "children": [
          1,
          19,
        ],
        "title": "Functions",
      },
    ],
    "id": 0,
    "kind": 1,
    "kindString": "Project",
    "name": "default",
    "originalName": undefined,
    "sources": undefined,
  },
  "typedocId": "default",
}
`;
